graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+from typing import Union, Tuple
+from .base import BaseVN
+from ..embed import BaseSpectralEmbed
+from ..embed import AdjacencySpectralEmbed as ase, LaplacianSpectralEmbed as lse
+import numpy as np
+from scipy.spatial import distance
+
+
+class SpectralVertexNominator(BaseVN):
+    """
+    Class for spectral vertex nomination on a single graph.
+
+    Given a graph G=(V,E) and a subset of V called S (the "seed"),
+    Single Graph Vertex Nomination is the problem of ranking all V
+    in order of relation to members of S.
+    Spectral Vertex Nomination solves this problem by embedding G
+    into a low dimensional euclidean space using any of the embedding
+    algorithms (:ref:`tutorials <embed_tutorials>`),
+    and then generating a nomination list by some distance based algorithm.
+    In the simplest case, for each seed vertex u, the other vertices are
+    ranked in order of euclidean distance from u.
+    There can be both attributed and unattributed cases. In the unattributed
+    case, we treat each seed vertex individually and rank all vertices by distance.
+    In the attributed case, subsets of the seed vertices share some attribute we
+    care about. We want to rank the other vertices by the likelihood that they
+    also share that attribute. Note that the unattributed problem is simply the
+    attributed problem when each seed vertex has a unique attribute.
+    SVN algorithms in general make the assumption that vertexes
+    are likely to be related in a way that is of interest if they are close to
+    each other in an embedding. This is a somewhat simplified notion of relatedness,
+    and is not appropriate for all Vertex Nomination problems.
+
+    Parameters
+    ----------
+    embedding: np.ndarray, default = None
+        A pre-calculated embedding may be provided, in which case
+        it will be used for vertex nomination instead of embedding
+        the adjacency matrix using embeder.
+    embeder: str or BaseEmbed, default = 'ASE'
+        May provide either a embed object or a string indicating
+        which embedding method to use, which may be either
+        "ASE" for :py:class:`~graspologic.embed.AdjacencySpectralEmbed` or
+        "LSE" for :py:class:`~graspologic.embed.LaplacianSpectralEmbed`.
+    persistent : bool, default = True
+        If ``False``, future calls to fit will overwrite an existing embedding. Must be ``True``
+        if an embedding is provided.
+
+
+    Attributes
+    ----------
+    embedding : np.ndarray
+        The spectral embedding of the graph that spectral nomination will be preformed on.
+    embeder : :py:class:`~graspologic.embed.BaseEmbed`
```

Comment:
I put it in attributes because the value of `embedder` might change during execution. I'm not really sure if that means it should be included in the attributes section, should I still remove it?

Reply:
where does it change during execution? Regardless, i don't think it's necessary since someone can call `get_params`